File: Programming Language/Python3/maximum-twin-sum-of-a-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def pairSum(self, head: Optional[ListNode]) -> int:
        n = 0
        tem = head
        tail = None
        while tem != None:
            newNode = ListNode(tem.val)
            if tail is None:
                tail = newNode
            else:
                newNode.next = tail
                tail = newNode
            n += 1
            tem = tem.next
        ans = 0
        for i in range(0, n//2):
            ans = max(ans, head.val+tail.val)
            head = head.next
            tail = tail.next
        return ans

File: Programming Language/Java/maximum-twin-sum-of-a-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public int pairSum(ListNode head) {
       int len = 0;
       ListNode tem = head;
       ListNode tail = null;
       while(tem != null){
           ListNode newNode = new ListNode(tem.val);
           if(tail == null){
               tail = newNode;
           }else{
               newNode.next = tail;
               tail = newNode;
           }
           len++;
           tem = tem.next;
       } 
       int ans = 0;
       for(int i = 0; i < len/2; i++){
           ans = Math.max(ans, head.val+tail.val);
           head = head.next;
           tail = tail.next;
       }
       //System.out.print(len);
       return ans;
    }
}
